contest-service/service.bal

```diff
@@ -84,6 +154,109 @@ service /contestService on new http:Listener(9098) {
         }
     }
 
+    resource function get contests/report/[string contestId]() 
+            returns http:InternalServerError|string[][]{
+        stream<SubmissionData, persist:Error?> submissionstream = self.db->/submissions;
+        string[][]|persist:Error csvContentdata = from var submission in submissionstream
+            where submission.contestId == contestId
+            select [submission.id, submission.challenge.id, submission.challenge.title, submission.user.id, submission.user.username, submission.user.fullname, timeToString(submission.submittedTime), submission.score.toString()];
+        if csvContentdata is persist:Error {
+            log:printError("Error while reading submission data", 'error = csvContentdata);
+            return <http:InternalServerError>{
+                body: {
+                    message: string `Error while reading submission data`
+                }
+            };
+        } else {
+        string[] headers = ["submissionId", "challengeId", "challageTitle", "userId", "userName", "fullname", "submittedTime", "score"];
+        string[][] csvData= [headers, ...csvContentdata];
+        return csvData;
+        }
+    }
+
+    resource function get contests/[string userId]/registered(string status)
```

Comment:
Let's remove the status query param, as discussed. It's best to reduce calls from UI to backend.